src/txn/txn.c

```diff
@@ -826,43 +827,49 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 		 * For prepared transactions commit timestamp could be earlier
 		 * than stable timestamp.
 		 */
-		WT_ERR(__wt_txn_commit_timestamp_validate(
```

Comment:
Could this function call `__wt_txn_set_timestamp` instead of parsing the timestamps out here and duplicating the logic? Let's talk about that tomorrow.

Reply:
Thank you @agorrod for quick review. I would like your suggestion on whether commit should return an error if prepare timestamp is set for a non-prepared transaction, or should we clear the prepare timestamp? I am noting down here so as not to forget to discuss this.